[BUG] After the update to cognito stack (version 2.x.x for the constructs) there's an error

**Describe the bug**
Executing the wcdevsdeployer as it was working before, now, when the cognitostack is tried to be deployed, there's the following error:
https://github.com/lealceldeiro/org.wcdevs.blog.awsdeployer/runs/4289714989?check_suite_focus=true

```
Error:  Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:java (default-cli) on project awsdeployer: An exception occured while executing the Java class. Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
Error:  Error: Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
Error:      at new Construct (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/core/lib/construct-compat.js:43:19)
Error:      at new AwsCustomResource (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.js:156:9)
Error:      at /tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:58
Error:      at Kernel._wrapSandboxCode (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8860:24)
Error:      at Kernel._create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:34)
Error:      at Kernel.create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8173:29)
Error:      at KernelHost.processRequest (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9757:36)
Error:      at KernelHost.run (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9720:22)
Error:      at Immediate._onImmediate (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9721:46)
Error:      at processImmediate (internal/timers.js:464:21)
Error:  -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:java (default-cli) on project awsdeployer: An exception occured while executing the Java class. Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
Error: Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
    at new Construct (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/core/lib/construct-compat.js:43:19)
    at new AwsCustomResource (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.js:156:9)
    at /tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:58
    at Kernel._wrapSandboxCode (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8860:24)
    at Kernel._create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:34)
    at Kernel.create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8173:29)
    at KernelHost.processRequest (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9757:36)
    at KernelHost.run (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9720:22)
    at Immediate._onImmediate (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9721:46)
    at processImmediate (internal/timers.js:464:21)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: software.amazon.jsii.JsiiException: Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
Error: Cannot use tokens in construct ID: describeUserPool${Token[TOKEN.190]}
    at new Construct (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/core/lib/construct-compat.js:43:19)
    at new AwsCustomResource (/tmp/jsii-kernel-0MhyiA/node_modules/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.js:156:9)
    at /tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:58
    at Kernel._wrapSandboxCode (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8860:24)
    at Kernel._create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8432:34)
    at Kernel.create (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:8173:29)
    at KernelHost.processRequest (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9757:36)
    at KernelHost.run (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9720:22)
    at Immediate._onImmediate (/tmp/jsii-java-runtime15176792737432388773/lib/program.js:9721:46)
    at processImmediate (internal/timers.js:464:21)
    at software.amazon.jsii.JsiiRuntime.processErrorResponse (JsiiRuntime.java:124)
    at software.amazon.jsii.JsiiRuntime.requestResponse (JsiiRuntime.java:95)
    at software.amazon.jsii.JsiiClient.createObject (JsiiClient.java:89)
    at software.amazon.jsii.JsiiEngine.createNewObject (JsiiEngine.java:603)
    at software.amazon.awscdk.customresources.AwsCustomResource.<init> (AwsCustomResource.java:30)
    at software.amazon.awscdk.customresources.AwsCustomResource$Builder.build (AwsCustomResource.java:264)
    at org.wcdevs.blog.cdk.CognitoStack.userPoolClientSecretValue (CognitoStack.java:280)
    at org.wcdevs.blog.cdk.CognitoStack.createUserPoolClientSecret (CognitoStack.java:231)
    at org.wcdevs.blog.cdk.CognitoStack.lambda$newInstance$0 (CognitoStack.java:88)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:195)
    at java.util.AbstractList$RandomAccessSpliterator.forEachRemaining (AbstractList.java:720)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:484)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:474)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:497)
    at org.wcdevs.blog.cdk.CognitoStack.newInstance (CognitoStack.java:87)
    at org.wcdevs.blog.awsdeployer.CognitoDeployer.main (CognitoDeployer.java:59)
    at org.codehaus.mojo.exec.ExecJavaMojo$1.run (ExecJavaMojo.java:254)
    at java.lang.Thread.run (Thread.java:829)
```

**To Reproduce**
Steps to reproduce the behavior:
1. Use version 2.0.1 from cdk-constructs
2. Deploy the required construct to deploy the cognito stack
3. Deploy cognito stack
4. See error

**Expected behavior**
To get the cognitostack deployed properly

**OS (please complete the following information):**
 - GitHub environment: ubuntu-lates

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
   <groupId>org.wcdevs.blog</groupId>
   <artifactId>cdk</artifactId>
-  <version>2.0.2-SNAPSHOT</version>
+  <version>2.0.2</version>
   <packaging>jar</packaging>
 
   <name>wcDevs.org AWS CDK Constructs</name>
@@ -53,7 +53,7 @@
     <developerConnection>scm:git:https://github.com/lealceldeiro/org.wcdevs.blog.cdk.git
     </developerConnection>
     <url>https://github.com/lealceldeiro/org.wcdevs.blog.cdk</url>
-    <tag>org.wcdevs.blog.cdk-${project.version}</tag>
+    <tag>cdk-2.0.2</tag>
   </scm>
 
   <distributionManagement>

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
   <groupId>org.wcdevs.blog</groupId>
   <artifactId>cdk</artifactId>
-  <version>2.0.2</version>
+  <version>2.0.3-SNAPSHOT</version>
   <packaging>jar</packaging>
 
   <name>wcDevs.org AWS CDK Constructs</name>
@@ -53,7 +53,7 @@
     <developerConnection>scm:git:https://github.com/lealceldeiro/org.wcdevs.blog.cdk.git
     </developerConnection>
     <url>https://github.com/lealceldeiro/org.wcdevs.blog.cdk</url>
-    <tag>cdk-2.0.2</tag>
+    <tag>org.wcdevs.blog.cdk-${project.version}</tag>
   </scm>
 
   <distributionManagement>

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
   <groupId>org.wcdevs.blog</groupId>
   <artifactId>cdk</artifactId>
-  <version>2.0.1</version>
+  <version>2.0.2-SNAPSHOT</version>
   <packaging>jar</packaging>
 
   <name>wcDevs.org AWS CDK Constructs</name>
@@ -53,7 +53,7 @@
     <developerConnection>scm:git:https://github.com/lealceldeiro/org.wcdevs.blog.cdk.git
     </developerConnection>
     <url>https://github.com/lealceldeiro/org.wcdevs.blog.cdk</url>
-    <tag>cdk-2.0.1</tag>
+    <tag>org.wcdevs.blog.cdk-${project.version}</tag>
   </scm>
 
   <distributionManagement>

--- a/src/main/java/org/wcdevs/blog/cdk/CognitoStack.java
+++ b/src/main/java/org/wcdevs/blog/cdk/CognitoStack.java
@@ -229,7 +229,7 @@ public final class CognitoStack extends Stack {
                                                    String userPoolClientId, String clientName,
                                                    String secretName) {
     var userPoolClientSecretValue = userPoolClientSecretValue(scope, awsRegion, userPoolId,
-                                                              userPoolClientId);
+                                                              userPoolClientId, clientName);
     var secretTpl = String.format("{\"%s\": \"%s\",\"%s\": \"%s\",\"%s\": \"%s\",\"%s\": \"%s\"}",
                                   USER_POOL_ID_HOLDER, userPoolId,
                                   USER_POOL_CLIENT_ID_HOLDER, userPoolClientId,
@@ -252,7 +252,7 @@ public final class CognitoStack extends Stack {
   }
 
   private static String userPoolClientSecretValue(Stack scope, String awsRegion, String userPoolId,
-                                                  String userPoolClientId) {
+                                                  String userPoolClientId, String clientName) {
     // The UserPoolClient secret, can't be accessed directly
     // This custom resource will call the AWS API to get the secret,
     // See: https://github.com/aws/aws-cdk/issues/7225
@@ -270,8 +270,7 @@ public final class CognitoStack extends Stack {
     var policy = AwsCustomResourcePolicy.fromSdkCalls(SdkCallsPolicyOptions.builder()
                                                                            .resources(ANY_RESOURCE)
                                                                            .build());
-    var id = "describeUserPool" + userPoolClientId;
-    var userPoolResource = AwsCustomResource.Builder.create(scope, id)
+    var userPoolResource = AwsCustomResource.Builder.create(scope, "describeUserPool" + clientName)
                                                     .resourceType(resourceType)
                                                     .installLatestAwsSdk(false)
                                                     .onUpdate(userPoolClientMetadata)

